modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -41,7 +43,20 @@ def run_plan(scope, plan_name, named_args = {})
         old_run_as = executor.run_as
         executor.run_as = run_as
       end
-      result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, use_args, true)
+      if named_args['_catch_errors']
+        begin
+          result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, use_args, true)
```

Comment:
What? This is long... if it doesn't make sense for whatever it's calling to provide a more readable signature, I'd opt to put it in its own function that has a more readable name.

Reply:
I guess this was here before. But since we're calling it twice, I'd opt for taking this opportunity to make it more readable.

A separate concern is that we could simplify this by putting the `_catch_errors` check in the rescue block and reraise if false, but I'm comfortable avoiding it completely as well.

Reply:
In the end I missed an ensure around the run_as here so I've moved everything into the same block. I think the backtraces might be worse but the line number is probably more useful anyway.